spark/src/main/scala/io/delta/tables/DeltaColumnBuilder.scala

```diff
@@ -96,6 +99,69 @@ class DeltaColumnBuilder private[tables](
     this
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Specify a column as an identity column that is always generated by the system, i.e., does not
+   * allow user specified values using a start of 1 and a step of 1.
+   *
+   * @since 3.2.0
+   */
+  @Evolving
+  def generatedAlwaysAsIdentity(): DeltaColumnBuilder = {
+    generatedAlwaysAsIdentity(start = 1, step = 1)
+  }
```

Comment:
Could we maybe use constants for the default start and step? 
We could then use it in spark/src/test/scala/org/apache/spark/sql/delta/DDLTestUtils.scala instead of hard-coding the 1s.